lib/MySQL_HostGroups_Manager.cpp

```diff
@@ -8253,6 +8254,119 @@ void MySQL_HostGroups_Manager::HostGroup_Server_Mapping::remove_HGM(MySrvC* srv)
 	srv->ConnectionsFree->drop_all_connections();
 }
 
+/**
+* @brief Updates replication hostgroups by adding autodiscovered mysql servers.
+* @details Adds each server from 'new_servers' to the 'runtime_mysql_servers' table.
+* We then rebuild the 'mysql_servers' table as well as the internal 'hostname_hostgroup_mapping'.
+* @param new_servers A vector of tuples where each tuple contains the values needed to add each new server.
+*
+* @return Returns EXIT_FAILURE code on failure and EXIT_SUCCESS code on success.
+*/
+int MySQL_HostGroups_Manager::add_discovered_servers_to_mysql_servers_and_replication_hostgroups(vector<tuple<string, int, int>> new_servers) {
+	int exit_code = EXIT_SUCCESS;
+	bool added_new_server = false;
+	wrlock();
+
+	try {
+		for (tuple<string, int, int> s : new_servers) {
+			string host = std::get<0>(s);
+			uint16_t port = std::get<1>(s);
+			long int hostgroup_id = std::get<2>(s);
+
+			// Add the discovered server with default values
+			MySrvC* mysrvc = new MySrvC(
+				const_cast<char*>(host.c_str()), port, 0, 1, MYSQL_SERVER_STATUS_ONLINE, 0, -1, 0, -1, 0, const_cast<char*>("Discovered endpoint")
+			);
+			add(mysrvc, hostgroup_id);
+
+			proxy_info(
+				"Adding new discovered server %s:%d with: hostgroup=%d, weight=%ld, max_connections=%ld, use_ssl=%d\n",
+				host.c_str(), port, hostgroup_id, mysrvc->weight, mysrvc->max_connections, mysrvc->use_ssl
+			);
+
+			added_new_server = true;
+		}
+
+		if (added_new_server) {
+			purge_mysql_servers_table();
+			mydb->execute("DELETE FROM mysql_servers");
+			proxy_debug(PROXY_DEBUG_MYSQL_CONNPOOL, 4, "DELETE FROM mysql_servers\n");
+			generate_mysql_servers_table();
+			update_table_mysql_servers_for_monitor(false);
+			rebuild_hostname_hostgroup_mapping();
```

Comment:
With the previous suggested changes, this function should be replaced by `update_hostgroup_manager_mappings`. The two functions are identical, only differing in the missing check on `hgsm_mysql_servers_checksum` in the new `rebuild_hostname_hostgroup_mapping` function. Updating this checksums is addressed in the previous comment, as it's better for keeping general consistency. If I have not missed anything, I suggest deleting this new function.

Reply:
Deleting the new `rebuild_hostname_hostgroup_mapping` and using `update_hostgroup_manager_mappings` instead